test/runUnitTests.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bats
+
+NHC_DIR=$1
+logpath=/tmp/nhclogfile.log
+
+if [ -z "$NHC_PATH" ]; then
+   NHC_DIR=/opt/azurehpc/test/azurehpc-health-checks/
+fi
+
+NHC_PATH=$NHC_DIR/run-health-checks.sh
+
+get_sad_path_conf(){
+    SKU=$( curl -H Metadata:true --max-time 10 -s "http://169.254.169.254/metadata/instance/compute/vmSize?api-version=2021-01-01&format=text")
+    SKU="${SKU,,}"
+    if echo "$SKU" | grep -q "nd96asr_v4"; then
+        conf_name="nd96asr_v4"
+    elif echo "$SKU" | grep -q "nd96amsr_a100_v4"; then
+        conf_name="nd96amsr_a100_v4"
+    elif echo "$SKU" | grep -q "nd96isr_h100_v5"; then
+        conf_name="nd96isr_h100_v5"
+    elif echo "$SKU" | grep -q "hb176rs_v4"; then
+        conf_name="hb176rs_v4"
+    elif echo "$SKU" | grep -q "hx176rs"; then
+        conf_name="hx176rs"
+    else
+        echo "Unit-test for this SKU $SKU is not supported" 
+        return 1
+    fi
+    relative_path="$NHC_DIR/test/bad_test_confs/$conf_name.conf"
+    echo "$(realpath -m $relative_path)"
+    return 0
+}
+
+check_installed(){
+    if [ -x "$(command -v  nhc)" ]; then
+        #echo "NHC exists."
+        echo 0
+        return 0
+    else
+        echo "NHC executable does not exist."
+        return 1
+    fi
+}
+
+
+@test "Installation check" {
+    set +e
+    command -v  nhc
+    result=$?
+    set -e
+    [ "$result" -eq 0 ]
+}
+
+@test "Default checks Pass (Happy Path)" {   
+    sudo $NHC_PATH -o $logpath
+    result=$?
+    [ "$result" -eq 0 ]
+}
+
+@test "Checks adjusted to fail (Sad Path)" {
+    bad_conf_file=$(get_sad_path_conf)
+    if [[ "$bad_conf_file" == *"not supported"* ]]; then
+        return false
+    fi
+    set +e
```

Comment:
Why do we have to use set +e and -e for certain tests?

Reply:
This is specific to how BATs works. It 'sets -e' internally and this causes the script to exit if one of the scripts it calls fails. In the test cases we are making NHC fail, so if we don't set +e before hand it would exit the unit test script when we test a sad path.